tests/components/withings/conftest.py

```diff
@@ -100,33 +102,40 @@ def mock_config_entry(expires_at: int, scopes: list[str]) -> MockConfigEntry:
     )
 
 
-@pytest.fixture(name="setup_integration")
-async def mock_setup_integration(
-    hass: HomeAssistant, config_entry: MockConfigEntry
-) -> Callable[[], Coroutine[Any, Any, MockWithings]]:
-    """Fixture for setting up the component."""
-    config_entry.add_to_hass(hass)
+@pytest.fixture(name="withings")
+def mock_withings():
+    """Mock withings."""
 
-    assert await async_setup_component(hass, "application_credentials", {})
-    await async_import_client_credential(
-        hass,
-        DOMAIN,
-        ClientCredential(CLIENT_ID, CLIENT_SECRET),
-        DOMAIN,
+    mock = AsyncMock(spec=ConfigEntryWithingsApi)
+    mock.user_get_device.return_value = UserGetDeviceResponse(
+        **load_json_object_fixture("withings/get_device.json")
     )
-    await async_process_ha_core_config(
-        hass,
-        {"internal_url": "http://example.local:8123"},
+    mock.measure_get_meas.return_value = MeasureGetMeasResponse(
+        **load_json_object_fixture("withings/get_meas.json")
+    )
+    mock.sleep_get_summary.return_value = SleepGetSummaryResponse(
+        **load_json_object_fixture("withings/get_sleep.json")
     )
+    mock.notify_list.return_value = NotifyListResponse(
+        **load_json_object_fixture("withings/notify_list.json")
+    )
+
+    with patch(
+        "homeassistant.components.withings.common.ConfigEntryWithingsApi",
+        return_value=mock,
+    ):
+        yield mock
+
 
-    async def func() -> MockWithings:
-        mock = MockWithings()
-        with patch(
-            "homeassistant.components.withings.common.ConfigEntryWithingsApi",
-            return_value=mock,
-        ):
-            assert await async_setup_component(hass, DOMAIN, {})
-            await hass.async_block_till_done()
-        return mock
-
-    return func
+@pytest.fixture(name="disable_webhook_delay")
+def disable_webhook_delay():
+    """Disable webhook delay."""
+
+    mock = AsyncMock()
+    with patch(
+        "homeassistant.components.withings.common.SUBSCRIBE_DELAY", timedelta(seconds=0)
+    ), patch(
+        "homeassistant.components.withings.common.UNSUBSCRIBE_DELAY",
+        timedelta(seconds=0),
+    ):
+        yield mock
```

Comment:
Generally this is OK, but it would be better if you could use `async_fire_time_changed` instead

Reply:
Oh that would be a good idea as well! I'll do it after I've untangled everything and got the new lib in